### PR TITLE
Chore/code lint

### PR DESCRIPTION
**Pull request by Emmanuel-Develops**

### CODE REVIEW

**Comment by Extheoisah** [comment]
Is this ready for review?

File: src/service/api/search/useSearch.ts [hunk 1]
```diff
@@ -13,7 +13,7 @@ export const useSearch = ({
   filterFields,
   sortFields,
 }: SearchQuery) => {
-  const hasFilters = Boolean(filterFields.length);
+  // const hasFilters = Boolean(filterFields.length);
```
Comment (Extheoisah) [inline src/service/api/search/useSearch.ts:16]: Remove the comment if the code is not in use

File: src/components/customSearchbox/SearchBoxView.tsx [hunk 2]
```diff
@@ -1,3 +1,4 @@
+/* eslint-disable no-unused-vars */
```
Comment (Extheoisah) [inline src/components/customSearchbox/SearchBoxView.tsx:1]: Why do we need this global rule and not in the specific code block?

File: src/components/customSearchbox/SearchBox.tsx
Comment (Extheoisah) [inline src/components/customSearchbox/SearchBox.tsx:1]: A bunch of commented code in this file. Remove if not needed

**Comment by Emmanuel-Develops** [comment]
> Is this ready for review?

It is now!

File: src/components/customSearchbox/SearchBoxView.tsx (hunk 2 above)
Comment (Emmanuel-Develops) [reply src/components/customSearchbox/SearchBoxView.tsx]: there were some purposefully unused destructured variables, but I've added the rule to each of them instead.
in 814765a

File: .github/workflows/lint.yml [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+  pull_request:
+    branches: [main, staging]
+
+jobs:
+  lint:
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v2
```
Comment (Extheoisah) [inline .github/workflows/lint.yml:15]: Let's use v3 or v4 as v2 is getting depreciated

**Review by Extheoisah — APPROVED** [review]

**Pull request merged**